xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayASS.h

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "DVDOverlay.h"
+#include "cores/VideoPlayer/DVDSubtitles/SubtitlesAdapter.h"
+
+class CDVDOverlayASS : public CDVDOverlay
+{
+public:
+  explicit CDVDOverlayASS(std::shared_ptr<CSubtitlesAdapter> adapter)
+    : CDVDOverlay(DVDOVERLAY_TYPE_ASS)
+  {
+    replace = true;
+    m_adapter = std::move(adapter);
+  }
+
+  CDVDOverlayASS(CDVDOverlayASS& src) : CDVDOverlay(src), m_adapter(src.m_adapter) {}
+
+  ~CDVDOverlayASS() override = default;
+
+  CDVDOverlayASS* Clone() override { return new CDVDOverlayASS(*this); }
+
+  /*!
+  \brief Getter for the subtitle adapter
+  \return The subtitle adapter.
+  */
+  std::shared_ptr<CSubtitlesAdapter> GetSubtitlesAdapter() const { return m_adapter; }
```

Comment:
is just a class name not referring to the native/adapted type
which name do you prefer?

Reply:
Here, none. It should use CDVDLibass directly not the adapter.

Reply:
I thought it was good practice not to use the Libass class itself also here

Reply:
You only need the adapter for subtitle types that need to be adapted to ass. The reason being that if for some reason we need a different backend to process them in the future, you only need to change the adapter logic.

For ass based subs (parser and codec) you'll only ever have to change the code if you rely on another lib rather than libass.

Reply:
done